client/state/account-recovery/settings/selectors.js

```diff
@@ -0,0 +1,39 @@
+export const isAccountRecoverySettingsReady = ( state ) => {
+	return null != state.accountRecovery.settings.data;
+};
+
+const getSettingsData = ( state ) => {
+	return state.accountRecovery.settings.data;
+};
+
+export const isAccountRecoveryPhoneValidated = ( state ) => {
+	return isAccountRecoverySettingsReady( state ) ? getSettingsData( state ).phoneValidated : false;
+};
+
+export const isAccountRecoveryEmailValidated = ( state ) => {
+	return isAccountRecoverySettingsReady( state ) ? getSettingsData( state ).emailValidated : false;
+};
+
+export const getAccountRecoveryEmail = ( state ) => {
+	return isAccountRecoverySettingsReady( state ) ? getSettingsData( state ).email : '';
```

Comment:
`isAccountRecoverySettingsReady` is checking if `state.accountRecovery.settings.data` is null so that might actually not be a problem